core/src/main/scala/kafka/controller/PartitionStateMachine.scala

```diff
@@ -355,6 +355,7 @@ class ZkPartitionStateMachine(config: KafkaConfig,
     val (invalidPartitionsForElection, validPartitionsForElection) = leaderIsrAndControllerEpochPerPartition.partition { case (_, leaderIsrAndControllerEpoch) =>
       leaderIsrAndControllerEpoch.controllerEpoch > controllerContext.epoch
     }
+    //TODO: if we find any invalid partitions for election shouldn't the controller thread exit since there is a new controller?
```

Comment:
not sure if we want to do it inside PartitionStateMachine.scala, maybe the controller change event has already been en-queued.

Reply:
Removed TODO